graphql/types.go

```diff
@@ -43,6 +43,18 @@ func (e *Enum) enumValues() []string {
 	return e.Values
 }
 
+// CustomScalar is a leaf value with a custom Unwrapper function.
+type CustomScalar struct {
```

Comment:
if we go down this path, check that this implements `Type` (see line 106)

Reply:
It does! Otherwise we wouldn't be able to return it from the getType func.